crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -51,11 +51,18 @@ pub struct Interaction {
     pub internalize: bool,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LiquidityProvider {
+    Amm,
+    Pmm,
+}
+
 /// A uniswap pool deployed as part of the blockchain setup.
 #[derive(Debug, Clone, Copy)]
 pub struct Pool {
     pub reserve_a: Asset,
     pub reserve_b: Asset,
+    pub liquidity_provider: LiquidityProvider,
```

Comment:
This nesting doesn't seem coherent. `LiquidityProvider:PMM` doesn't seem to care about reserves. Shouldn't Pool become a higher level `Liquidity` enum which can be either a `Pool` or `PMM` (both of which can implement `Into<Liquidity>`

Reply:
Uniswap is still being used which requires reserves in both cases to get the env properly configured.

Reply:
Because on-chain interactions still use uniswap contract: https://github.com/cowprotocol/services/blob/2bf814045e45d723597bf3648d4a43a2fc4e075b/crates/driver/src/tests/setup/blockchain.rs#L642-L653

And to simulate this interaction, reserves are required:
https://github.com/cowprotocol/services/blob/2bf814045e45d723597bf3648d4a43a2fc4e075b/crates/driver/src/tests/setup/blockchain.rs#L424-L434

That was the simplest way to put it into practice. Right now, we're able to use any limit price and quote, but this means we have to adjust the reserve values to ensure the simulation works out. If the reserves ratio isn't in the right proportion to the quote amounts, it causes the simulation to fail.

Reply:
I see, but isn't the concept of PMM misleading in that case (it has nothing to do with a private market maker providing a quote).

Instead, could we just have a method to create a pool that satisfies a certain quote which is called in the main test case and leave the existing struct untouched?

Reply:
Yeah, now it makes sense to remove it. Had quite a lot of experiments, and post refactoring is still required. Replaced the `LiquidityProvider` struct with the quote.